frontend/components/Reports/FindingSuppression.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import api from '../../util/federalistApi';
+import notificationActions from '../../actions/notificationActions';
+
+function FindingSuppression({
+  ruleId = '',
+  suppressed = false,
+  suppressedBy = null,
+  siteId,
+  sbtId,
+  sbtType,
+  sbtCustomRules = [],
+}) {
+  const [customerRules, setCustomerRules] = useState(sbtCustomRules);
+  const willBeSuppressed = customerRules.find(r => r.id === ruleId);
+
+  useEffect(() => {
+    setCustomerRules(sbtCustomRules);
+  }, [sbtCustomRules]);
```

Comment:
I'm caveating this comment by saying we should probably investigate swapping in a consistent querying library like [`react-query`](https://tanstack.com/query/latest/docs/framework/react/overview) because I've also lost the plot a little on hook re-rending logic. But I think this might be the block that is causing you issues: I don't know that you need a `useEffect` call because `sbtCustomRules` are used to initialize the local state already. And they shouldn't change since we aren't refetching the report while on this page.

Reply:
So i deleted the useEffect, but I am not sure that the set state for customer rules is working... it seems to issue at the same time as the update function, so I just passed in the new array as a param instead. Is that bad? I feel like it needs a promise somewhere, but I haven't seen a pattern like that in how we manage updates